### ISSUE TITLE
Single collection of Users

### ISSUE THREAD

**Issue by shahnawab** [octobercms/october#4722]
I am using OctoberCMs for last few months and is amazed by ease of using it for development. 

One thing that I would like to implement is that single user base is there for frontend and backend. or to extend the backend features to frontend.

It will make work more easy and manageable. Currently we have to develop the same feature in frontend that has been developed in backend.

**Comment by SebastiaanKloos** [comment]
The reason why the two are seperated is that you won’t allow frontend users in your backend. However, I think it would be better if the two are the same tables and based on roles

**Comment by shahnawab** [comment]
Its making it coumbersome to manage frontend users with backend users.

While we have controls over Roles and Groups, why not merge both of them?

**Comment by bennothommo** [comment]
@shahnawab I believe the separation of Backend user and Front-end user (via the User plugin) is by design. There are many people who don't need user functionality on their front-end websites, which is why it is separated off to a plugin. It also puts a clear divide between what "visitors" of the front-end can access, versus what "administrators" of the back-end can access.

**Comment by LukeTowers** [comment]
@shahnawab there's lots of different ways to use October and the way that the user system is implement between the backend vs the frontend is highly dependent on the specific project in question. Most of the time if you find yourself re-implementing the entire backend in the frontend, what you typically should have done instead is just used the backend by itself and reskinned it to match your design. 

I've worked on several complex web applications that don't even include the CMS module, they just use the backend one. I've also worked on applications where it made sense to have backend and frontend users at the same time and re-implementing some of the backend features in the frontend, but it ended up that it would have just made more sense to handle that all in the backend to begin with.

Without more knowledge about how exactly you're using October and what for we can't provide any more specific suggestions for improving your development experience.

**Comment by LarBearrr** [comment]
I'm going to chime in here, mainly because I have developed both kinds of sites/apps with October; I can see both sides of this argument. 

**TL;DR:** Think about what it is you are hoping to accomplish by giving frontend users to the backend and remember you can impersonate frontend users from the backend. Some backend features are not hard to implement on the frontend. Be responsible about how you implement your requirements. 
<hr>

1. To @LukeTowers point, if you find yourself wanting to give frontend users access to the backend more often than not, it is worth exploring how using the backend can solve your problems.
2. Likewise, if you find yourself reaching for backend features on the frontend more often than not, the same suggestion from above applies. 
3. However, the level of control you have over the backend customization requires considerable effort and knowledge of October.
4. By implementing your requirements on the frontend, you truly have complete freedom to develop your site/app as needed, including the ability to use more modern stacks like Vue, etc. 
5. Finally, I would agree that it may be beneficial to allow backend users to sign in to the frontend. I believe this is available through impersonation which is implemented with `RainLab.Blog`: [L11-L19](https://github.com/rainlab/user-plugin/blob/bed2661ca4164fa21500d491b5ac0f9ebfed4b6f/controllers/users/_preview_toolbar.htm#L11-L19)

One of the paradoxes of October is how easy it is to build sites/apps because many common (and less common, but useful) features of modern web apps (forms, lists, CRUD, etc.) are basically available to us for free. As a result, I think we tend to become a little lazy when it comes to building out the public facing site/app and wish we could just "reuse" the backend feature on the frontend.

**Comment by CptMeatball** [comment]
Personally, I don' think those lists should match. Have you ever seen the `Users` section of a WordPress website that's got WooCommerce installed? It's absolutely maddening to say the least. 😱
